plugins/completion/coq-thirdparty.nix

```diff
@@ -0,0 +1,46 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.coq-thirdparty;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.coq-thirdparty = {
+    enable = mkEnableOption "coq-thirdparty";
+
+    package = helpers.mkPackageOption "coq-thirdparty" pkgs.vimPlugins.coq-thirdparty;
+
+    sources = mkOption {
+      type = types.listOf types.attrs;
```

Comment:
This is not really a random attribute set. Per the documentation it must have   'src' and optionally shortName. Then the options are source specific.

I'd see maybe something like a listOf {
   src = ...;
   shortName = ....;
   extraOptions  = ....;
}

In addition some sources may depend on external plugins, so we could analyze the sources and enable the right plugins when we find them ? Or is it going too far ?